compiler/transforms/clear_memory_space.py

```diff
@@ -8,28 +9,28 @@
 class ClearMemorySpace(ModulePass):
     name = "clear-memory-space"
 
-    def apply(self, ctx: MLContext, module: builtin.ModuleOp) -> None:
+    def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
         # helper function to clear the memory space of a memref
         # also clears the layout information of the memref - not used anymore
         def clear_memory_space(t):
-            if isinstance(t, memref.MemRefType):
+            if isinstance(t, builtin.MemRefType):
```

Comment:
I don't understand why this is necessary to change if we haven't updated xdsl

Reply:
It's not necessary, but memreftype is defined in `builtin`, not in `memref`. 
This is just to not have this message screaming at me:
![image](https://github.com/user-attachments/assets/10be5eb3-0fe9-485d-b27a-21677eba096c)

I know it's not really related to this PR, but I tend to just resolve these kinds of things when I notice them and then sneak them in

Reply:
Oh okay yeah sure no problem at all :)